src/test/java/com/pubnub/api/managers/SubscriptionManagerTest.java

```diff
@@ -1331,11 +1326,20 @@ public void file(@NotNull PubNub pubnub, @NotNull PNFileEventResult pnFileEventR
     @SuppressWarnings("unchecked")
     @Test
     public void testSubscribeBuilderWithState() throws PubNubException {
-        final String expectedPayload = PubNubUtil.urlDecode("%7B%22ch1%22%3A%5B%22p1%22%2C%22p2%22%5D%2C%22cg2%22%3A" +
-                "%5B%22p1%22%2C%22p2%22%5D%7D");
+        String p1Key = "p1";
```

Comment:
If you haven't touch state in Subscription manager why this test does it have to change?

Reply:
As we discussed yesterday.
When you execute setState(withHearbeat=false) REST that is being called returns 
"State must be a JSON object." in case state is not JSON object. 
When you execute setState(withHearbeat=true) REST that is being called doesn't returns error but also state is not being set.

I added validation to setState method:
```
if (!isJsonObject(stringifiedState)) {
            throw PubNubException.builder()
              .pubnubError(PubNubErrorBuilder.PNERROBJ_STATE_MUST_BE_JSON_OBJECT).build();
}
```
Now when validation is in place I had to change state value in test so it pass that validation.